### ISSUE TITLE
Cannot run tox for the first time with this plugin installed

### ISSUE THREAD

**Issue by thedrow** [tox-dev/tox-pipenv#5]
I'm not doing anything unusual. I just run tox with `pipenv run tox` and I get the following error:

```
GLOB sdist-make: /home/omer/Documents/Projects/myproject/setup.py
py36 create: /home/omer/Documents/Projects/myproject/.tox/py36
ERROR: Error creating virtualenv. Note that spaces in paths are not supported by virtualenv. Error details: FileNotFoundError(2, 'No such file or directory')
__________________________________________________________________________________ summary __________________________________________________________________________________
ERROR:   py36: Error creating virtualenv. Note that spaces in paths are not supported by virtualenv. Error details: FileNotFoundError(2, 'No such file or directory')
```

**Comment by tonybaloney** [comment]
Tox is the one that creates the pipenv environments, so the idea is just to run `tox` and it will create the virtualenv's using pipenv.

**Comment by thedrow** [comment]
This issue is about the fact that tox fails to create a virtual environment with this plugin installed.
If I run tox first without the plugin installed and then install the plugin and run tox again #6 occurs.

**Comment by WhyNotHugo** [comment]
Having the same issue:

```
$ tox -e py36
GLOB sdist-make: /home/hugo/workspace/Hugo/django-afip/setup.py
py36 create: /home/hugo/workspace/Hugo/django-afip/.tox/py36
ERROR: Error creating virtualenv. Note that spaces in paths are not supported by virtualenv. Error details: FileNotFoundError(2, 'No such file or directory')
___________________________________________________________ summary ___________________________________________________________
ERROR:   py36: Error creating virtualenv. Note that spaces in paths are not supported by virtualenv. Error details: FileNotFoundError(2, 'No such file or directory')
```

~However, if I run `mkdir .tox/py36` first, it works fine.~ Scrub that last bit, it still fails down the line. That just makes it skip venv creation.

**Comment by thedrow** [comment]
I just verified that the same behavior is reproduced with master.

**Comment by thedrow** [comment]
@tonybaloney I'd be happy to resolve this issue but I need a nudge in the right direction.
Is there any information you can provide which will help me resolve this problem?

**Comment by tonybaloney** [comment]
@thedrow creating an integration test was the first thing. Reproducing these issues is tricky.

Started on something here: https://github.com/tonybaloney/tox-pipenv/blob/master/test/integration/test.sh

**Comment by tonybaloney** [comment]
This is the output from the integration test FWIW 

```
Running virtualenv with interpreter /Users/anthonyshaw/repo/tox-pipenv/env/bin/python3.6
Using real prefix '/Library/Frameworks/Python.framework/Versions/3.6'
New python executable in /Users/anthonyshaw/repo/tox-pipenv/test/integration/env/bin/python3.6
Also creating executable in /Users/anthonyshaw/repo/tox-pipenv/test/integration/env/bin/python
Installing setuptools, pip, wheel...done.
Obtaining file:///Users/anthonyshaw/repo/tox-pipenv
Collecting tox>=2.9.1 (from tox-pipenv==1.3.0)
  Using cached tox-2.9.1-py2.py3-none-any.whl
Collecting pipenv==11.0.2 (from tox-pipenv==1.3.0)
  Downloading pipenv-11.0.2.tar.gz (4.1MB)
    100% |████████████████████████████████| 4.1MB 351kB/s
Collecting six (from tox>=2.9.1->tox-pipenv==1.3.0)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting py>=1.4.17 (from tox>=2.9.1->tox-pipenv==1.3.0)
  Using cached py-1.5.2-py2.py3-none-any.whl
Collecting virtualenv>=1.11.2; python_version != "3.2" (from tox>=2.9.1->tox-pipenv==1.3.0)
  Using cached virtualenv-15.1.0-py2.py3-none-any.whl
Collecting pluggy<1.0,>=0.3.0 (from tox>=2.9.1->tox-pipenv==1.3.0)
Collecting pew>=0.1.26 (from pipenv==11.0.2->tox-pipenv==1.3.0)
  Using cached pew-1.1.2-py2.py3-none-any.whl
Requirement already satisfied: pip>=9.0.1 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pipenv==11.0.2->tox-pipenv==1.3.0)
Collecting virtualenv-clone>=0.2.5 (from pew>=0.1.26->pipenv==11.0.2->tox-pipenv==1.3.0)
  Downloading virtualenv_clone-0.3.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=17.1 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pew>=0.1.26->pipenv==11.0.2->tox-pipenv==1.3.0)
Building wheels for collected packages: pipenv
  Running setup.py bdist_wheel for pipenv ... done
  Stored in directory: /Users/anthonyshaw/Library/Caches/pip/wheels/cf/d2/80/320f73eea7022843679a25ff1e1f8df1d0bfab9b4bee860c58
Successfully built pipenv
Installing collected packages: six, py, virtualenv, pluggy, tox, virtualenv-clone, pew, pipenv, tox-pipenv
  Running setup.py develop for tox-pipenv
Successfully installed pew-1.1.2 pipenv-11.0.2 pluggy-0.6.0 py-1.5.2 six-1.11.0 tox-2.9.1 tox-pipenv virtualenv-15.1.0 virtualenv-clone-0.3.0
GLOB sdist-make: /Users/anthonyshaw/repo/tox-pipenv/setup.py
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/python
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py36 inst-nodeps: /Users/anthonyshaw/repo/tox-pipenv/.tox/dist/tox-pipenv-1.3.0.zip
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py36 installed: pew==1.1.2,pipenv==11.0.2,pluggy==0.6.0,py==1.5.2,six==1.11.0,tox==2.9.1,tox-pipenv==1.3.0,virtualenv==15.1.0,virtualenv-clone==0.3.0
py36 runtests: PYTHONHASHSEED=None
py36 runtests: commands[0] | python -m pytest test/
=========================================================================================================== test session starts ============================================================================================================
platform darwin -- Python 3.6.1, pytest-3.4.1, py-1.5.2, pluggy-0.6.0
rootdir: /Users/anthonyshaw/repo/tox-pipenv, inifile:
collected 1 item

test/test_nothing.py .                                                                                                                                                                                                               [100%]

========================================================================================================= 1 passed in 0.01 seconds =========================================================================================================
_________________________________________________________________________________________________________________ summary __________________________________________________________________________________________________________________
  py36: commands succeeded
  congratulations :)
```

**Comment by tonybaloney** [comment]
And trying to repro by running `pipenv run tox`. There is maybe something environment specific as it "works on my machine" :-D

```
Already using interpreter /Users/anthonyshaw/repo/tox-pipenv/env/bin/python3.6
Using real prefix '/Library/Frameworks/Python.framework/Versions/3.6'
New python executable in /Users/anthonyshaw/repo/tox-pipenv/test/integration/env/bin/python3.6
Also creating executable in /Users/anthonyshaw/repo/tox-pipenv/test/integration/env/bin/python
Installing setuptools, pip, wheel...done.
Obtaining file:///Users/anthonyshaw/repo/tox-pipenv
Requirement already satisfied: tox>=2.9.1 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox-pipenv==1.3.0)
Requirement already satisfied: pipenv==11.0.2 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox-pipenv==1.3.0)
Requirement already satisfied: virtualenv>=1.11.2; python_version != "3.2" in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox>=2.9.1->tox-pipenv==1.3.0)
Requirement already satisfied: six in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox>=2.9.1->tox-pipenv==1.3.0)
Requirement already satisfied: py>=1.4.17 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox>=2.9.1->tox-pipenv==1.3.0)
Requirement already satisfied: pluggy<1.0,>=0.3.0 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from tox>=2.9.1->tox-pipenv==1.3.0)
Requirement already satisfied: pew>=0.1.26 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pipenv==11.0.2->tox-pipenv==1.3.0)
Requirement already satisfied: pip>=9.0.1 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pipenv==11.0.2->tox-pipenv==1.3.0)
Requirement already satisfied: setuptools>=17.1 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pew>=0.1.26->pipenv==11.0.2->tox-pipenv==1.3.0)
Requirement already satisfied: virtualenv-clone>=0.2.5 in /Users/anthonyshaw/repo/tox-pipenv/env/lib/python3.6/site-packages (from pew>=0.1.26->pipenv==11.0.2->tox-pipenv==1.3.0)
Installing collected packages: tox-pipenv
  Found existing installation: tox-pipenv 1.3.0
    Uninstalling tox-pipenv-1.3.0:
      Successfully uninstalled tox-pipenv-1.3.0
  Running setup.py develop for tox-pipenv
Successfully installed tox-pipenv
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project.
GLOB sdist-make: /Users/anthonyshaw/repo/tox-pipenv/setup.py
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/python
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py27
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py27 inst-nodeps: /Users/anthonyshaw/repo/tox-pipenv/.tox/dist/tox-pipenv-1.3.0.zip
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py27
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py27
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py27 installed: pew==1.1.2,pipenv==11.0.2,pluggy==0.6.0,py==1.5.2,six==1.11.0,tox==2.9.1,tox-pipenv==1.3.0,virtualenv==15.1.0,virtualenv-clone==0.3.0
py27 runtests: PYTHONHASHSEED=None
py27 runtests: commands[0] | python -m pytest test/
=========================================================================================================== test session starts ============================================================================================================
platform darwin -- Python 2.7.10, pytest-3.3.2, py-1.5.2, pluggy-0.6.0
rootdir: /Users/anthonyshaw/repo/tox-pipenv, inifile:
collected 1 item

test/test_nothing.py .                                                                                                                                                                                                               [100%]

========================================================================================================= 1 passed in 0.01 seconds =========================================================================================================
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/python
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py36 inst-nodeps: /Users/anthonyshaw/repo/tox-pipenv/.tox/dist/tox-pipenv-1.3.0.zip
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
py36 installed: pew==1.1.2,pipenv==11.0.2,pluggy==0.6.0,py==1.5.2,six==1.11.0,tox==2.9.1,tox-pipenv==1.3.0,virtualenv==15.1.0,virtualenv-clone==0.3.0
py36 runtests: PYTHONHASHSEED=None
py36 runtests: commands[0] | python -m pytest test/
=========================================================================================================== test session starts ============================================================================================================
platform darwin -- Python 3.6.1, pytest-3.4.1, py-1.5.2, pluggy-0.6.0
rootdir: /Users/anthonyshaw/repo/tox-pipenv, inifile:
collected 1 item

test/test_nothing.py .                                                                                                                                                                                                               [100%]

========================================================================================================= 1 passed in 0.01 seconds =========================================================================================================
lint recreate: /Users/anthonyshaw/repo/tox-pipenv/.tox/lint
lint installdeps: pep8
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/python
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/lint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
lint inst: /Users/anthonyshaw/repo/tox-pipenv/.tox/dist/tox-pipenv-1.3.0.zip
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/lint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/lint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
lint installed: pew==1.1.2,pipenv==11.0.2,pluggy==0.6.0,py==1.5.2,six==1.11.0,tox==2.9.1,tox-pipenv==1.3.0,virtualenv==15.1.0,virtualenv-clone==0.3.0
lint runtests: PYTHONHASHSEED=None
lint runtests: commands[0] | flake8 --ignore=E402 --max-line-length=100 tox_pipenv/
tox_pipenv/plugin.py:19:1: E302 expected 2 blank lines, found 1
ERROR: InvocationError: '/Users/anthonyshaw/repo/tox-pipenv/env/bin/python3.6 -m pipenv run flake8 --ignore=E402 --max-line-length=100 tox_pipenv/'
pylint recreate: /Users/anthonyshaw/repo/tox-pipenv/.tox/pylint
pylint installdeps: pylint
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/python
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/pylint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
pylint inst: /Users/anthonyshaw/repo/tox-pipenv/.tox/dist/tox-pipenv-1.3.0.zip
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/pylint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
WARNING:test command found but not installed in testenv
  cmd: /Users/anthonyshaw/repo/tox-pipenv/env/bin/pip
  env: /Users/anthonyshaw/repo/tox-pipenv/.tox/pylint
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
pylint installed: pew==1.1.2,pipenv==11.0.2,pluggy==0.6.0,py==1.5.2,six==1.11.0,tox==2.9.1,tox-pipenv==1.3.0,virtualenv==15.1.0,virtualenv-clone==0.3.0
pylint runtests: PYTHONHASHSEED=None
pylint runtests: commands[0] | pylint -E --rcfile=./.pylintrc tox_pipenv/
Traceback (most recent call last):
  File "/Users/anthonyshaw/repo/tox-pipenv/.tox/pylint/.venv/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File "/Users/anthonyshaw/repo/tox-pipenv/.tox/pylint/.venv/lib/python3.6/site-packages/pylint/__init__.py", line 16, in run_pylint
    Run(sys.argv[1:])
  File "/Users/anthonyshaw/repo/tox-pipenv/.tox/pylint/.venv/lib/python3.6/site-packages/pylint/lint.py", line 1301, in __init__
    linter.read_config_file()
  File "/Users/anthonyshaw/repo/tox-pipenv/.tox/pylint/.venv/lib/python3.6/site-packages/pylint/config.py", line 646, in read_config_file
    raise IOError("The config file {:s} doesn't exist!".format(config_file))
OSError: The config file ./.pylintrc doesn't exist!
ERROR: InvocationError: '/Users/anthonyshaw/repo/tox-pipenv/env/bin/python3.6 -m pipenv run pylint -E --rcfile=./.pylintrc tox_pipenv/'
_________________________________________________________________________________________________________________ summary __________________________________________________________________________________________________________________
  py27: commands succeeded
  py36: commands succeeded
ERROR:   lint: commands failed
ERROR:   pylint: commands failed
```

**Comment by thedrow** [comment]
With tox-pipenv 1.3.0 I get the following error when I run `pipenv run tox` or `pipenv shell; tox`:
```
Traceback (most recent call last):
  File "/home/omer/.pyenv/versions/3.6.4/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/omer/.pyenv/versions/3.6.4/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/__main__.py", line 4, in <module>
    cli()
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 1063, in invoke
    Command.invoke(self, ctx)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/vendor/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/cli.py", line 49, in cli
    from . import core
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/core.py", line 33, in <module>
    from .project import Project
  File "/home/omer/.local/share/virtualenvs/infraboost--RkZn49j/lib/python3.6/site-packages/pipenv/project.py", line 31, in <module>
    raise RuntimeError('Given PIPENV_PIPFILE is not found!')
RuntimeError: Given PIPENV_PIPFILE is not found!
```
This issue must have been a problem with a previous pipenv version.

**Comment by thedrow** [comment]
And now that I removed .tox again I'm getting the original error. So no dice there :(

**Comment by thedrow** [comment]
I'm currently debugging this and I get:
`FileNotFoundError: [Errno 2] No such file or directory: '/home/omer/Documents/Projects/myproject/.tox/py36/Pipfile'` because venv.path points to `/home/omer/Documents/Projects/myproject/.tox/py36/` and the Pipfile is not present.
For some reason the entire error is not printed to the screen and so we get something like https://github.com/tonybaloney/tox-pipenv/issues/5#issue-287102044.

**Comment by tonybaloney** [comment]
released in 1.4.0
